### PR TITLE
[RFR] Store form in state rather than props

### PR DESCRIPTION
**Pull request by fzaninotto**
Not quite sure of myself. @djhi and @jpetitcolas, can you review?

### CODE REVIEW

File: src/mui/list/FilterForm.js [hunk 1]
```diff
@@ -35,7 +48,7 @@ export class FilterForm extends Component {
                             name={filterElement.props.source}
                             component={filterElement.type}
                             resource={resource}
-                            record={currentFilters}
+                            record={this.state.currentFilters}
```
Comment (djhi) [inline src/mui/list/FilterForm.js]: deconstruct as for props

**Comment by djhi** [comment]
Not sure I understand how it solves the filter problem.

**Comment by djhi** [comment]
btw, it does not work at all locally, can't set any filters anymore:
http://screencloud.net/v/y10w

**Comment by fzaninotto** [comment]
It doesn't fix the filter problem. Since it also adds problems on your side (that I can't reproduce), let's forget it.

**Pull request closed without merging**